ext/proxy/Dxxxx.md

```diff
@@ -877,207 +890,155 @@ Change concept `Readable` ([readable.iterators]) as follows:
->     StrictWeakOrder<as_function_t<F>, iter_common_reference_t<I1>, iter_common_reference_t<I2>>();
+>     StrictWeakOrder<F, value_type_t<I1>, value_type_t<I2>>() &&
+>     StrictWeakOrder<F, value_type_t<I1>, reference_t<I2>>() &&
+>     StrictWeakOrder<F, reference_t<I1>, value_type_t<I2>>() &&
+>     StrictWeakOrder<F, reference_t<I1>, reference_t<I2>>() &&
+>     StrictWeakOrder<F, iter_common_reference_t<I1>, iter_common_reference_t<I2>>();
+> }
+>
+> template <class> struct indirect_result_of { };
+> template <class F, class... Is>
+>   requires IndirectCallable<F, Is...>()
+> struct indirect_result_of<F(Is...)> :
+>   result_of<F(reference_t<Is>...)> { };
+> ```
+
+Note: These definitions of `IndirectCallable` and `IndirectCallablePredicate` are less general than the ones in N4569 that they replace. The original definitions are variadic but these handle only up to 2 arguments. The Standard Library never requires callback functions to accept more than two arguments, so the reduced expressive power does not impact the Standard Library; however, it may impact user code. The complication is the need to check callability with a cross-product of the parameters' `value_type_t` and `reference_t`s, which is difficult to express using Concepts and results in an explosion of tests to be performed as the number of parameters increases.
+
+There are several options for preserving the full expressive power of the N4569 concepts should that prove desirable: (1) Require callability testing only with arguments "`value_type_t<Is>...`", "`reference_t<Is>...`" , and "`iter_common_reference_t<Is>...`", leaving the other combinations as merely documented constraints that are not required to be tested; (2) Actually test the full cross-product of argument types using meta-programming techniques, accepting the compile-time hit when argument lists get large. (The latter has been tested and shown to be feasable.)
+
+Change the definition of the `projected` struct ([projected]) to the following:
+
+> ```c++
+> template <Readable I, IndirectRegularCallable<I> Proj>
+> struct projected {
+>   using value_type = decay_t<indirect_result_of_t<Proj&(I)>>;
+>   indirect_result_of_t<Proj&(I)> operator*() const;
+> };
+> ```
+
+
+Remove the definition of the `rvalue_reference_t` type alias in  `IndirectlyMovable` ([commonalgoreq.indirectlymovable]).
+
+Change the description of the `IndirectlyMovableStorable` concept ([commonalgoreq.indirectlymovable]) to be:
+
+> 3\. Let `i` be a dereferenceable object of type `In`,
+> let `o` be a dereferenceable object of type `Out`. Then
+> `IndirectlyMovableStorable<In, Out>()` is satisfied if
+> and only if
+>
+> (3.1) -- If `Out` satisfies `Readable`, the expression `*o`
+> has the same value after either `*o = iter_move(i)` or 
+> `*o = value_type_t<In>(iter_move(i))`.
+
+Change the description of the `IndirectlyCopyableStorable` concept ([commonalgoreq.indirectlycopyable]) to be:
```

Comment:
Same here: both `value_type_t<In> v = *i` and `*o = v` are individually required by the algorithms and must be supported by `IndirectlyCopyableStorable`; the semantics of the composition of the two expressions is derivable.